tests/unit/ShareStructure.spec.ts

```diff
@@ -0,0 +1,393 @@
+    inputElement.setValue(200)
+    inputElement.trigger('change')
+    await wrapper.vm.$nextTick()
+    await flushPromises()
+    await wrapper.vm.$nextTick()
+    expect(wrapper.find(formSelector).text())
+      .toContain('The number for the series (or all series combined, if there are multiple under ' +
+      'a class) cannot exceed the number for the class')
+    expect(wrapper.vm.$data.formValid).toBe(false)
+    wrapper.destroy()
+  })
+
+  it('Shows error message if sum of series maximum shares is greater than class max shares', async () => {
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 200, true, 0.50, 'CAD', true)
+    const shareSeries1 = createShareStructure(1, 1, 'Series', 'Series A', true, 100, true, 0.50, 'CAD', true)
+    const shareSeries2 = createShareStructure(1, 1, 'Series', 'Series B', true, 50, true, 0.50, 'CAD', true)
+    shareClass.series.push(shareSeries1)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareSeries2, 0, -1, 0, [shareClass])
+    const inputElement: Wrapper<Vue> = wrapper.find(txtMaxShares)
+    inputElement.setValue(150)
+    inputElement.trigger('change')
+    await wrapper.vm.$nextTick()
+    await flushPromises()
+    await wrapper.vm.$nextTick()
+    expect(wrapper.find(formSelector).text())
+      .toContain('The number for the series (or all series combined, if there are multiple under ' +
+      'a class) cannot exceed the number for the class')
+    expect(wrapper.vm.$data.formValid).toBe(false)
+    wrapper.destroy()
+  })
+
+  it('Shows series par value and currency in read only mode', async () => {
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
+    const shareSeries = createShareStructure(1, 1, 'Series', 'Series A', true, 100, true, 0.50, 'CAD', true)
+    shareClass.series.push(shareSeries)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareSeries, 0, -1, 0, [shareClass])
+    expect((<HTMLInputElement> wrapper.find(seriesParValue).element).value)
+      .toEqual(shareSeries['parValue'].toString())
+    expect(wrapper.find(seriesParValue).attributes('disabled')).toBe('disabled')
+    expect((<HTMLInputElement> wrapper.find(seriesCurrency).element).value)
+      .toEqual('Canadian dollar (CAD)')
+    expect(wrapper.find(seriesCurrency).attributes('disabled')).toBe('disabled')
+    wrapper.destroy()
+  })
+
+  it('Do not display series no max share option if class has max share', async () => {
+    const shareClass = createShareStructure(null, 1, 'Class', 'Class A', true, 100, true, 0.50, 'CAD', true)
+    const shareSeries = createShareStructure(1, 1, 'Series', 'Series A', true, 100, true, 0.50, 'CAD', true)
+    shareClass.series.push(shareSeries)
+    const wrapper: Wrapper<ShareStructure> = createComponent(shareSeries, 0, -1, 0, [shareClass])
+    expect(wrapper.find('#lbl-no-maximum').exists()).toBe(false)
+    wrapper.destroy()
+  })
+})
```

Comment:
Need a test that changing currency (eg, to USD) works (and keeps working).

Reply:
Added a test for this